Source/DataSources/GeometryVisualizer.js

```diff
@@ -152,17 +162,28 @@ define([
 
         var numberOfShadowModes = ShadowMode.NUMBER_OF_SHADOW_MODES;
         this._outlineBatches = new Array(numberOfShadowModes);
-        this._closedColorBatches = new Array(numberOfShadowModes);
-        this._closedMaterialBatches = new Array(numberOfShadowModes);
-        this._openColorBatches = new Array(numberOfShadowModes);
-        this._openMaterialBatches = new Array(numberOfShadowModes);
+        this._closedColorBatches = new Array(numberOfShadowModes * 3);
+        this._closedMaterialBatches = new Array(numberOfShadowModes * 3);
+        this._openColorBatches = new Array(numberOfShadowModes * 3);
+        this._openMaterialBatches = new Array(numberOfShadowModes * 3);
```

Comment:
I feel like this batching is starting to get out of hand (even when it was just shadows), but probably worth addressing later.